api/prometheus/v1/api.go

```diff
@@ -824,6 +824,7 @@ type apiOptions struct {
 
 type Option func(c *apiOptions)
 
+// WithTimeout can be used to provide an optional query evaluation timeout for Query and QueryRange.
```

Comment:
Let's add a link to API docs to give users more context on this

Reply:
@kakkoyun Done 👍